docs/proposals/11-public-prefix.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Refactor Prefix type for IPv6 usage
+
+iep-number: 11
+
+creation-date: 2024-02-25
+
+status: implementable
+
+authors:
+
+- "@afitzler"
+
+reviewers:
+
+- "@guvenc"
+- "@MalteJ"
+
+---
+
+# IEP-11: Refactor Prefix type for IPv6 usage
+
+## Table of Contents
+
+- [IEP-11: Public Prefix](#iep-11-public-prefix)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+  - [Alternatives](#alternatives)
+
+## Summary
+
+In an IPv6 environment, dynamically allocating IP prefixes is crucial for efficient address space management, ensuring 
+network security, and maintaining scalability. Allowing users to freely choose IPs and prefixes can lead to conflicts, 
+inefficient use of the address space, and security vulnerabilities. Furthermore, the use of local ranges contradicts 
+the design principles of IPv6, aimed at eliminating the need for such practices by providing a vast address space. 
+Therefore, we propose a new approach to how users can assign prefixes to `NetworkInterfaces`, as the current model, 
+introduced by the [Networking Integration IEP](01-networking-integration.md), does not align with these principles.
+
+## Motivation
+
+Currently, the `NetworkInterface` allows defining `IPs` and `Prefixes` for both IP families (IPv4 and IPv6) either by 
+providing a discrete value or by using an `EphemeralPrefixSource`. The `EphemeralPrefixSource` can be derived from an 
+IPv4 or IPv6 `Prefix`, allowing for the free selection of `NetworkInterface` IPs. In the IPv4 realm, it is common 
+practice to use a local, non-publicly routable IP range, with internet connectivity ensured via a `NATGateway` for 
+egress and a `LoadBalancer` for ingress traffic. However, this approach is not applicable to IPv6.
+Here we need to ensure that either a unique local IPv6 address is being used (`fd00::/8`) or an IP or `Prefix` is derived 
```

Comment:
@MalteJ Do we want to support this non-routable range ? In the same VNI, the VMs can communicate with each other.